Interior refactor: Aragog, Zalmoxis, atmodeller, dummy modules, testing overhaul, version pinning, oxygen accounting (#678)
FormingWorlds/PROTEUS: docs/Reference/config/atmosphere.md

# Atmosphere and chemistry

The `[atmos_clim]` section configures the atmospheric climate module (radiative-convective
structure and surface energy balance). The `[atmos_chem]` section configures
atmospheric chemistry (photochemical kinetics).

Submodule documentation:

> [AGNI](https://www.h-nicholls.space/AGNI/) |

Reviewer: ```suggestion
[AGNI](https://www.h-nicholls.space/AGNI/) 
[SOCRATES](https://proteus-framework.org/SOCRATES/) 
```

Author: Applied.

[JANUS](https://proteus-framework.org/JANUS/) |
[VULCAN](https://github.com/FormingWorlds/VULCAN).
See also [Model description](../../Explanations/model.md#atmosphere-climate-agni-janus).

## Atmosphere climate `[atmos_clim]`

| Parameter | Type | Default | Description |
|-----------|------|---------|-------------|

> | `module` | str | `"agni"` | Climate module: `agni` (detailed RCE solver), `janus` (fast grey/correlated-k), `dummy` (parameterised) |

Reviewer: ```suggestion
| `module` | str | `"agni"` | Climate module: `agni` (radiative-convective atmosphere model), `janus` (prescribed pseudoadiabatic model) `dummy` (parameterised) |
```

Author: Applied.

> | `spectral_group` | str | `"Honeyside"` | Opacity k-table set (shared by AGNI and JANUS) |

Reviewer: ```suggestion
| `spectral_group` | str | `"Honeyside"` | Opacity k-table set (shared by AGNI and JANUS; see [options here](https://proteus-framework.org/SOCRATES/Reference/proteus_spectral_file_reference.html)) |
```

Author: Applied.

| `surf_greyalbedo` | float | `0.1` | Grey surface albedo (used when `agni.surf_material = "greybody"`) |

**Surface boundary condition**

| Parameter | Type | Default | Description |
|-----------|------|---------|-------------|
| `surf_state` | str | `"skin"` | Energy balance scheme: `mixed_layer` (ocean-like), `fixed` (lock to magma temperature), `skin` (conductive skin layer) |
| `surface_d` | float | `0.01` | Conductive skin thickness [m] |
| `surface_k` | float | `2.0` | Conductive skin conductivity [W m$^{-1}$ K$^{-1}$] |

**Solver limits**

| Parameter | Type | Default | Description |
|-----------|------|---------|-------------|
| `tmp_minimum` | float | `0.5` | Temperature floor [K] |

### AGNI `[atmos_clim.agni]`

AGNI is a detailed radiative-convective equilibrium solver written in Julia.
It iteratively solves for the atmospheric temperature-pressure profile that
balances absorbed stellar radiation with outgoing thermal emission and
interior heat flux.

**Physics**

| Parameter | Type | Default | Description |
|-----------|------|---------|-------------|
| `solve_energy` | bool | `true` | Iterative energy-conserving T(p) profile |

> | `convection` | bool | `true` | Convective adjustment (mixing length theory) |

Reviewer: ```suggestion
| `convection` | bool | `true` | Include mixing-length convective energy fluxes |
```

Author: Applied.

| `real_gas` | bool | `false` | Real-gas EOS where available |
| `chemistry` | str or none | `"none"` | Atmospheric chemistry: `none` or `eq` (FastChem equilibrium) |
| `mlt_criterion` | str | `"s"` | Convection criterion: `s` (Schwarzschild) or `l` (Ledoux) |

**Surface properties**

| Parameter | Type | Default | Description |
|-----------|------|---------|-------------|
| `surf_material` | str | `"greybody"` | Surface scattering model; `"greybody"` uses `surf_greyalbedo` |
| `surf_roughness` | float | `1e-3` | Surface roughness scale [m] |
| `surf_windspeed` | float | `2.0` | Characteristic wind speed [m s$^{-1}$] |

**Condensation**

| Parameter | Type | Default | Description |
|-----------|------|---------|-------------|
| `phs_timescale` | float | `1e6` | Phase change relaxation timescale [s] |
| `evap_efficiency` | float | `0.01` | Raindrop re-evaporation efficiency [0, 1] |

**Solver tuning**

| Parameter | Type | Default | Description |
|-----------|------|---------|-------------|
| `solution_atol` | float | `0.5` | Newton solver absolute tolerance [W m$^{-2}$] |
| `solution_rtol` | float | `0.15` | Newton solver relative tolerance |

> | `psurf_thresh` | float | `0.1` | Skip full RCE when surface pressure is below this [bar] |

Reviewer: ```suggestion
| `psurf_thresh` | float | `0.1` | Switch to transparent atmosphere-free solver when surface pressure is less than this [bar] |
```

Author: Applied.
